src/metabase/api/card.clj

```diff
@@ -930,4 +932,48 @@ saved later when it is ready."
       (persisted-info/mark-for-pruning! {:id (:id persisted-info)} "off")
       api/generic-204-no-content)))
 
+(s/defn param-values
+  "Fetch values for a parameter.
+
+  The source of values could be:
+  - static-list: user defined values list
+  - card: values is result of running a card"
+  ([card param-key]
+   (param-values card param-key nil))
+
+  ([card      :- su/Map
+    param-key :- su/NonBlankString
+    query     :- (s/maybe su/NonBlankString)]
+   (let [param       (get (m/index-by :id (:parameters card)) param-key)
+         source-type (:values_source_type param)]
+     (when-not param
+       (throw (ex-info (tru "Card does not have a parameter with the ID {0}" (pr-str param-key))
+                       {:status-code 400})))
+     (case source-type
+       "static-list" (params.static-values/param->values param query)
+       "card"        (params.card-values/param->values param query)
+       (throw (ex-info (tru "Invalid values-source-type: {0}" (pr-str source-type))
+                       {:values-source-type source-type
+                        :status-code        400}))))))
+
+(api/defendpoint GET "/:card-id/params/:param-key/values"
+  "Fetch possible values of the parameter whose ID is `:param-key`.
+
+    ;; fetch values for Card 1 parameter 'abc' that are possible
+    GET /api/card/1/params/abc/values"
+  [card-id param-key]
+  (let [card (api/read-check Card card-id)]
+    (param-values card param-key)))
+
+(api/defendpoint GET "/:card-id/params/:param-key/search/:query"
```

Comment:
Why two api endpoints here? I would think that we could do something like
`GET api/<card-id>/params/<param-key>/values?search=foo` to handle both? As far as I can tell from the bodies of these, they are the same except for the inclusion or not of `query`. Would it make sense to collapse these into one and just check for a `?search=<search-query>` query param?

Reply:
this is to be asymmetry with ho`api/dashboard`. we could use one single API if you think it makes more sense.

Reply:
Would be happy to have a single endpoint instead of a pair. We could also consolidate  `api/dashboard/:dashId/params/:paramId/values` and `search` endpoints into just one endpoint.